src/Game.tsx

```diff
@@ -1,36 +1,97 @@
-export function Game({ pokemons }: Props): JSX.Element {
-  const [ballCoords, setBallCoods] = useState<Coords>({ x: 50, y: 50 });
-  const [player1Coord, setPlayer1Coord] = useState(0);
-  const [player2Coord, setPlayer2Coord] = useState(100);
-
-  function changePlayerCoords(e: KeyboardEvent): void {
-    console.log(e.key);
-    // Change state here
-    setBallCoods({ x: 100, y: 100 });
-    setPlayer1Coord(50);
-    setPlayer2Coord(50);
-  }
-  useEffect(() => {
-    document.addEventListener('keydown', changePlayerCoords);
+export function Game({ pokemonOne, pokemonTwo }: Props): JSX.Element {
+  const [ballCoordY] = useState(50);
+  const [ballCoordX] = useState(50);
+  const [player1Coord, setPlayer1Coord] = useState(50);
+  const [player2Coord, setPlayer2Coord] = useState(50);
+  const [isGameStarted, setIsGameStarted] = useState(false);
+  const interval = 40;
+  const step = 5;
+
+  const generateHandlePlayCoordsCallbacks = useCallback(() => {
+    let firstPlayerInt: TimeoutResult = null;
+    let secondPlayerInt: TimeoutResult = null;
+
+    const cleanHandle = () => {
+      if (firstPlayerInt) clearInterval(firstPlayerInt);
+      firstPlayerInt = null;
+      if (secondPlayerInt) clearInterval(secondPlayerInt);
+      secondPlayerInt = null;
+    };
+
+    const handleKeyDown = (e: KeyboardEvent) => {
+      switch (e.key) {
+        case 'w':
+        case 'W':
+          firstPlayerInt =
+            firstPlayerInt ||
+            setInterval(() => {
+              setPlayer1Coord((player1Coord) =>
+                player1Coord > 0 ? player1Coord - step : player1Coord
+              );
+            }, interval);
 
-    return () => {
-      document.removeEventListener('keydown', changePlayerCoords);
+          break;
+        case 's':
+        case 'S':
+          firstPlayerInt =
+            firstPlayerInt ||
+            setInterval(() => {
+              setPlayer1Coord((player1Coord) =>
+                player1Coord < 100 ? player1Coord + step : player1Coord
+              );
+            }, interval);
+          break;
+        case 'ArrowUp':
+          secondPlayerInt =
+            secondPlayerInt ||
+            setInterval(() => {
+              setPlayer2Coord((player2Coord) =>
+                player2Coord > 0 ? player2Coord - step : player2Coord
+              );
+            }, interval);
+          break;
+        case 'ArrowDown':
+          secondPlayerInt =
+            secondPlayerInt ||
+            setInterval(() => {
+              setPlayer2Coord((player2Coord) =>
+                player2Coord < 100 ? player2Coord + step : player2Coord
+              );
+            }, interval);
+          break;
+        default:
+          break;
+      }
     };
+
+    return { handleKeyDown, cleanHandle };
   }, []);
 
-  if (pokemons.includes('')) return <Redirect to="/" />;
+  useEffect(() => {
+    if (isGameStarted) {
+      const { cleanHandle, handleKeyDown } =
+        generateHandlePlayCoordsCallbacks();
```

Comment:
Having fewer hooks is almost always better because hooks aren't free and imply memory and CPU resources usage.
That is why I recommend moving `generateHandlePlayCoordsCallbacks` code directly into this `useEffect`.